src/formState.ts

```diff
@@ -476,7 +476,7 @@ function newObjectState<T, P = any>(
         throw new Error(`${key || "formState"} is currently readOnly`);
       }
       getFields(this).forEach((field) => {
-        if (field.key in value && (!field.dirty || !field._focused)) {
+        if (field.key in value) {
```

Comment:
Makes sense to remove this here and keep this logic isolated only in the `field.set` method.